app/src/main/java/org/tokend/template/features/invites/view/InviteNewUsersActivity.kt

```diff
@@ -103,34 +90,41 @@ class InviteNewUsersActivity : BaseActivity() {
         }
     }
 
+    private var inviteDisposable: Disposable? = null
     private fun invite() {
-        invitees?.let {
-            InviteNewUsersUseCase(
-                    it,
-                    walletInfoProvider,
-                    apiProvider,
-                    repositoryProvider
-            )
-                    .perform()
-                    .compose(ObservableTransformers.defaultSchedulersCompletable())
-                    .doOnSubscribe {
-                        isLoading = true
-                    }
-                    .doOnTerminate {
-                        isLoading = false
-                    }
-                    .subscribeBy(
-                            onComplete = this::onInviteComplete,
-                            onError = { error ->
-                                errorHandlerFactory.getDefault().handle(error)
-                            }
-                    )
-                    .addTo(compositeDisposable)
+        val invitees = this.invitees ?: return
+
+        val progress = ProgressDialogFactory.getDialog(this) {
+            inviteDisposable?.dispose()
         }
+
+        inviteDisposable?.dispose()
+        inviteDisposable = InviteNewUsersUseCase(
+                invitees,
+                walletInfoProvider,
+                apiProvider,
+                repositoryProvider
+        )
+                .perform()
+                .compose(ObservableTransformers.defaultSchedulersCompletable())
+                .doOnSubscribe {
+                    progress.show()
+                }
+                .doOnEvent {
+                    progress.hide()
```

Comment:
`.hide()` makes dialog invisible but not closed so it causes leaked window exception on activity finish